resourcename/validate.go

```diff
@@ -30,3 +34,51 @@ func Validate(name string) error {
 	}
 	return nil
 }
+
+// ValidatePattern that a resource name pattern conforms to the restrictions outlined in AIP-122,
```

Comment:
The comment for this function seems wrong. It should not mention names, but patterns.

Reply:
It does say `resource name pattern` - what do you think should be updated?

Reply:
Missed that. But it still only mentions DNS names, while it actually checks many more things. Either list everything, or include nothing imo.

Reply:
Will remove!